source/includes/code-examples/GridFS.cs

```diff
@@ -0,0 +1,415 @@
+        var bucket = new GridFSBucket(database);
+
+        // Uploads data to a stream, then writes the stream to a GridFS file
+        // start-upload-from-stream-async
+        using (var fileStream = new FileStream("/path/to/input_file", FileMode.Open, FileAccess.Read))
+        {
+            await bucket.UploadFromStreamAsync("new_file", fileStream);
+        }
+        // end-upload-from-stream-async
+    }
+
+    static void UploadStream()
+    {
+        // Initialize MongoDB client
+        var client = new MongoClient("<connection string>");
+        var database = client.GetDatabase("db");
+
+        // Creates a GridFS bucket or references an existing one
+        var bucket = new GridFSBucket(database);
+
+        // Uploads data to a stream, then writes the stream to a GridFS file
+        // start-upload-from-stream
+        using (var fileStream = new FileStream("/path/to/input_file", FileMode.Open, FileAccess.Read))
+        {
+            bucket.UploadFromStream("new_file", fileStream);
+        }
+        // end-upload-from-stream
+    }
+
+    static void Find()
+    {
+        // Initialize MongoDB client
+        var client = new MongoClient("<connection string>");
+        var database = client.GetDatabase("db");
+
+        // Creates a GridFS bucket or references an existing one
+        var bucket = new GridFSBucket(database);
+
+        // Prints information about each file in the bucket
+        // start-find
+        var filter = Builders<GridFSFileInfo>.Filter.Empty;
+
+        var files = bucket.Find(filter);
+
+        foreach (var file in files.ToEnumerable())
+        {
+            Console.WriteLine(file.ToJson());
+        }
+        // end-find
+    }
+
+    static async Task FindAsync()
+    {
+        // Initialize MongoDB client
+        var client = new MongoClient("<connection string>");
+        var database = client.GetDatabase("db");
+
+        // Creates a GridFS bucket or references an existing one
+        var bucket = new GridFSBucket(database);
+
+        // Prints information about each file in the bucket
+        // start-find-async
+        var filter = Builders<GridFSFileInfo>.Filter.Empty;
+
+        var files = await bucket.FindAsync(filter);
+
+        await files.ForEachAsync(file => { Console.WriteLine(file.ToJson()); });
+        // end-find-async
+    }
+
+    static async Task DownloadFileAsync()
+    {
+        // Initialize MongoDB client
+        var client = new MongoClient("<connection string>");
+        var database = client.GetDatabase("db");
+
+        // Creates a GridFS bucket or references an existing one
+        var bucket = new GridFSBucket(database);
+
+        // Downloads a file from the GridFS bucket by referencing its ObjectId value
+        // start-open-download-stream-async
+        var filter = Builders<BsonDocument>.Filter.Eq("filename", "new_file");
+        var doc = await database.GetCollection<BsonDocument>("fs.files").Find(filter).FirstOrDefaultAsync();
```

Comment:
Shouldn't we be using the `bucket.Find` method instead of accessing the underlying `fs.files` collection directly?

Same comment for other places doing the same thing.

Reply:
I agree that makes more sense. (This code was AI-translated from C++.)

Reply:
Waiting for the change to review it.